### PR TITLE
Replaces binding of compartment ids for std::array<uint64_t,2>

### PR DESCRIPTION
**Pull request by sergiorg-hpc**
Using `py-bluepy`, we observed recently that the `ids` of a compartment report were mapped into a Python list of tuples instead of returning a pointer to the actual data in the bindings. This PR solves this issue by correctly returning an `numpy.ndarray`, which can be used automatically by `py-bluepy` without further changes.

For instance, retrieving a single row of a reference dataset that contains close to 80 million columns per timestep, takes the following:
```
Without fix: 147.01s
With fix: 13.44s
```
Any of the operations that involve the `ids` incur in large data conversions, increasing the execution times. One example is `len(view.ids)` inside `py-bluepy`, that required close to 30 seconds.

Further details of the on-going investigations are available in HPCTM-1487.

Co-authored with @jorblancoa, @matz-e, and @alkino.

### CODE REVIEW

**Review by matz-e — DISMISSED** [review]

**Comment by sergiorg-hpc** [comment]
Pinging @mgeplf for approval and merge.

**Comment by matz-e** [comment]
I'm just going to add some performance numbers from @sergiorg-hpc:
image

**Comment by mgeplf** [comment]
FYI: I'll be looking at this in the next couple days.

File: include/bbp/sonata/common.h
Comment (mgeplf) [inline include/bbp/sonata/common.h:43]: can we add a comment here saying that this should be pairs of `<typeof(NodeID), typeof(ElementID)>`, but that we're using the extra space such that the bindings are (much!) faster?

**Review by mgeplf — COMMENTED** [review]
Wow, this is great; thanks for the investigation!

File: include/bbp/sonata/common.h
Comment (sergiorg-hpc) [reply include/bbp/sonata/common.h:45]: Done!

**Comment by sergiorg-hpc** [comment]
Thank you @matz-e for adding the info, and also @mgeplf for the comments and review.

If you guys agree, I'd suggest to merge this after the CI passes and create a new tag release for `libsonata`. I already have the PR in Spack if I'm not mistaken, so it should be a matter to either point to the specific commit after the merge (for temporary tags) or to the specific version tag (for formal releases).

@mgeplf, I believe you are the only one of us with permission to merge, so it is your call.

**Review by mgeplf — APPROVED** [review]

**Comment by mgeplf** [comment]
nice work, always happy w/ perf!

**Pull request merged**
